src/skdh/completeness/complete.py

```diff
@@ -0,0 +1,127 @@
+import os
+import pickle
+import numpy as np
+from skdh.base import BaseProcess
+from skdh.completeness.utils import load_subject_data, compute_completeness_master, compute_summary_metrics, check_hyperparameters
+from skdh.completeness.visualizations import visualize_overview_plot, plot_completeness, plot_data_gaps, plot_timescale_completeness
+
+
+class completeness_pipe(BaseProcess):
+    r"""
+    Pipeline for assessing signal completeness.
+    """
+
+    def __init__(
+            self,
+            subject,
+            subject_folder,
+            device_name,
+            fpath_output,
+            measures,
+            resample_width_mins=5,
+            gap_size_mins=30,
+            ranges=None,
+            data_gaps=None,
+            time_periods=None,
+            timescales=None):
+
+        check_hyperparameters(subject,
+            subject_folder,
+            device_name,
+            fpath_output,
+            measures,
+            resample_width_mins,
+            gap_size_mins,
+            ranges,
+            data_gaps,
+            time_periods,
+            timescales)
+
+        super().__init__(
+            subject=subject,
+            subject_folder=subject_folder,
+            device_name=device_name,
+            fpath_output=fpath_output,
+            measures=measures,
+            resample_width=resample_width_mins,
+            gap_size_mins=gap_size_mins,
+            ranges=ranges,
+            data_gaps=data_gaps,
+            time_periods=time_periods,
+            timescales=timescales)
+
+        self.subject = subject
+        self.subject_folder = subject_folder
+        self.device_name = device_name
+        self.fpath_output = fpath_output
+        self.measures = measures
+        self.resample_width_mins = resample_width_mins
+        self.gap_size_mins = gap_size_mins
+        self.ranges = ranges
+        self.data_gaps = data_gaps
+        self.time_periods = time_periods
+        self.timescales = timescales
+
+#    @handle_process_returns(results_to_kwargs=True)
+    def predict(self,
+                generate_figures=True,
+                **kwargs):
+        """
+        Compute completeness and save results. Create and save figures if generate_figures is True.
+        """
+        super().predict(
+            expect_days=False,
+            expect_wear=False,
+            **kwargs,
+        )
+        data_dic = load_subject_data(self.subject_folder, self.subject, self.device_name, self.measures, self.ranges)
```

Comment:
it would be better if we split loading data into its own function, and left completeness just to check any available data passed into the completeness module

Reply:
Done, divided up the processing in 4 steps: init pipe, load subject data, predict, generate_figures